### ISSUE TITLE
LW_ERROR_PASSWORD_MISMATCH

### ISSUE THREAD

**Issue by dgroos** [BeyondTrust/pbis-open#79]
**Version**: 8.5.4.334
**OS/Distro**: Ubuntu desktop 16.04 - unity desktop
**Issue/Impact**: Can't join domain

PBIS has been working fine since I finally got it working around 6 weeks ago. About a week ago I used clonezilla to clone this machine on which I run pbis, I refer to it as 'myserver' though it is running Ubuntu desktop. After cloning onto another computer, I successfully ran the cli-join command on that clone (though there was an issue on the clone yet to be solved, but that isn't the problem I'm reporting here).

When I replaced the clone with my regular computer (myserver), re-ran the cli-join command, much to my surprise I got the 'password_mismatch' error detailed below. I've spent several hours exploring this problem eg reading all issue reports here, eg #74 which has a similar error message but different symptoms, #27 which seems more similar though not quite, also tried all of the investigative/solving ideas given to no avail, and #22 leading me to try both removing samba and adding the lines listed in that issue report to the global section. I've also looked at several other pages I've googled with similar error messages, all to no avail. 
Any ideas?
Thanks,
David
PS My best guess at the moment comes from this line in the syslog (also given below) which says it's using my **computer** username, not my **AD** username:

> Nov  3 08:02:11 myserver lsass: [LwKrb5GetTgtImpl /builder/src-git/Platform/src/linux/lwadvapi/threaded/krbtgt.c:346] KRB5 Error code: -1765328360 (Message: Preauthentication failed)
Nov  3 08:02:11 myserver lsass: [lsass] The cached machine account password was rejected by the DC.
Nov  3 08:02:11 myserver lsass: [lsass] Failed to validate user for login (name = 'GIVEN IS MY SERVER USERNAME, NOT MY AD USERNAME') -> error = 40022, symbol = LW_ERROR_PASSWORD_MISMATCH, client pid = 5851

● lwsmd.service - BeyondTrust PBIS Service Manager
>   Loaded: loaded (/lib/systemd/system/lwsmd.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2017-11-02 16:14:59 CDT; 7h ago
  Process: 962 ExecStart=/opt/pbis/sbin/lwsmd --start-as-daemon (code=exited, status=0/SUCCESS)
 Main PID: 971 (lwsmd)
   CGroup: /system.slice/lwsmd.service
           ├─ 971 /opt/pbis/sbin/lwsmd --start-as-daemon
           ├─ 993 lw-container lwreg
           ├─1020 lw-container eventlog
           ├─1099 lw-container netlogon
           ├─1125 lw-container lwio
           ├─1151 lw-container lsass
           └─1181 lw-container reapsysl

> (repeated several times:) Nov 02 23:00:59 myserver lsass[1151]: [LwKrb5GetTgtImpl /builder/src-git/Platform/src/linux/lwadvapi/threaded/krbtgt.c:346] KRB5 Error code: -1765328360 (Message: Preauthentication failed)
Nov 02 23:00:59 myserver lsass[1151]: [lsass] The cached machine account password was rejected by the DC.

 2. /opt/pbis/bin/lwsm list
> lwreg          running (container: 993)
dcerpc         stopped
eventlog       running (container: 1020)
lsass          running (container: 1151)
lwio           running (container: 1125)
netlogon       running (container: 1099)
rdr            running (io: 1125)
reapsysl       running (container: 1181)
usermonitor    stopped

 3. /opt/pbis/bin/domainjoin-cli query
> Error: LW_ERROR_ACCESS_DENIED [code 0x00009cde]
Incorrect access attempt
 
4. pbis status
>I subbed in (my domain) etc for [site-specific info](http://paste.ubuntu.com/25915299/)

 5. /opt/pbis/bin/enum-users
[if I used "sudo..." on these commands, then command returned nothing and back to prompt]
> Failed to enumerate users.  Error code 40022 (LW_ERROR_PASSWORD_MISMATCH).
The password is incorrect for the given account

 6. attach logs
  - /opt/pbis/bin/lwsm set-log-target -p lsass -  file /tmp/lsass.log
> Error: LW_ERROR_ACCESS_DENIED (40158)
Incorrect access attempt

  - /opt/pbis/bin/lwsm set-log-level -p lsass - debug
> Error: LW_ERROR_ACCESS_DENIED (40158)
Incorrect access attempt

  - attach log
In syslog I get this repeated many dozens of times each our:
> Nov  3 08:02:11 myserver lsass: [LwKrb5GetTgtImpl /builder/src-git/Platform/src/linux/lwadvapi/threaded/krbtgt.c:346] KRB5 Error code: -1765328360 (Message: Preauthentication failed)
Nov  3 08:02:11 myserver lsass: [lsass] The cached machine account password was rejected by the DC.
Nov  3 08:02:11 myserver lsass: [lsass] Failed to validate user for login (name = '**NOTE:THIS IS MY SERVER USERNAME, NOT MY AD USERNAME**') -> error = 40022, symbol = LW_ERROR_PASSWORD_MISMATCH, client pid = 5851

**Output/Error**:
Shown above

**Steps to Reproduce**:
1. Domainjoin command
> sudo /opt/pbis/bin/domainjoin-cli join MYDOMAIN ADusername
Error: LW_ERROR_PASSWORD_MISMATCH [code 0x00009c56]

> The password is incorrect for the given account

[and just saw that you fixed the domainjoin-gui and ran that. The popup says that status was "Joined" (but still can't authenticate at clients) and when I pressed the 'Leave Domain' button I got: ERROR_ACCESS_DENIED message.]

**Comment by RBoulton-BT** [comment]
It's saying the "The cached machine account password was rejected by the DC." which looks like the PBIS agent does not have the correct password for the computer account in AD. 

Cloning the system may well have resulted in 2 agents using the same computer account in AD. The computer account password is changed during a join and periodically after that. Please ensure each machine is using a different computer account in AD (domainjoin-cli query), and then try leaving and re-joining to reset the password.

**Comment by dgroos** [comment]
Thanks for quick reply! I'm a teacher and today don't have students so have most all of the day to try to solve this issue.

To do as you suggested, I re-connected up my cloned computer, changed the hostname back to the original computers name (the one from which I had cloned it), rebooted and successfully issued the join and then the leave commands. Then I shut that clone down then booted the computer (which provided the image for the clone--the computer I'm trying to get working again). Was that your suggestion?

Unfortunately, upon issuing the join-cli command, I still get the error:

> Error: LW_ERROR_PASSWORD_MISMATCH [code 0x00009c56]
The password is incorrect for the given account

I wonder if there is a different cache I need to clear?

**Comment by RBoulton-BT** [comment]
Did you say the domainjoin-cli leave gives an error too?

**Comment by dgroos** [comment]
On the cloned machine (not the one I'm currently troubleshooting) I successfully joined and left the domain.

On the master computer (the one that provided the image, the one I'm trying to get to work again) I get the same error message trying to join or leave.

**Comment by RBoulton-BT** [comment]
Odd the leave shouldn't need to communicate with AD unless you pass a username. Could you try running:

domainjoin-cli leave --keepLicense

**Comment by dgroos** [comment]
Get same results w/that command.

Now might be a good time to bring up how I ended up solving #60 as shown on this comment: https://github.com/BeyondTrust/pbis-open/issues/60#issuecomment-331686391

I checked the lsassd.reg file and strangely the line which I had edited had returned to its default setting of:
> "ActiveDirectoryPasswordPrompt" = {
default = ""

So I wonder, if that setting had gotten reset, I wonder if any other settings might have reset to default? [Thanks so much for your quick support! Just to let you know I've got a meeting and will be returning in about 2 hours so will be off-line... :-( ]

**Comment by RBoulton-BT** [comment]
I just tried reproducing this locally. If I reset my AD computer password I get very similar results to you. However is I restart lsass my domainjoin-cli starts working correctly again. Could you try:
/opt/pbis/bin/lwsm restart lsass
domainjoin-cli leave
domain join-cli join <your domain etc>

**Comment by ParadingLunatic** [comment]
Wow, that fixed my problem that I mentioned on #74 where I couldn't even leave the domain as I was getting the password mismatch when trying to leave and join.  I even rebooted the laptop and still couldn't leave.  After doing the lwsm restart lsass it fixed my problem.

**Comment by dgroos** [comment]
SUCCESS
love seeing when pbis prints that on the screen :-) 
I did as you suggested and got the hoped-for results!
Thanks @RBoulton-BT for your close attention to this issue!
Am now working to get the clients to authenticate. Running into problem on #60 but solution described in https://github.com/BeyondTrust/pbis-open/issues/60#issuecomment-331686391  isn't working but am still exploring.
Again thanks for your help,
David

**Comment by dgroos** [comment]
About authentication not working on client... Continuing to explore problem. Previous solution from #60 doesn't seem to have traction. Looking at the [lsass.log](http://paste.ubuntu.com/25881802/) from an unsuccessful client I'm wondering about a lot of the "lw" entries I see. I assume those are left over from the "likewise" days? At 1 point I know I added a legacy packet in search of domainjoin-gui and am wondering if there might be some problem with that? Wondering if I should open another issue and put this, there?
Thanks,
David

**Comment by dgroos** [comment]
And finally, it works.
Not sure exactly why it finally does work, but it does. I applied basically the same strategy previously described in #60 however this time I had to change the default "" to "Password: " --this is different from #60 because at that point I could do any entry e.g. "passwerd" or whatever and it would work. Oh well, I'm sure there is a reason. Again, thanks for the help, my students have their computers back!
